Sources/Rosalind/Rosalind.swift

```diff
@@ -1 +1,88 @@
+import Command
+import CryptoKit
+import FileSystem
+import Foundation
+import Path
 
+enum RosalindError: LocalizedError {
+    case notFound(AbsolutePath)
+
+    var errorDescription: String? {
+        switch self {
+        case let .notFound(path):
+            return "File not found at path \(path.pathString)"
+        }
+    }
+}
+
+public protocol Rosalindable {
+    func analyze(path: AbsolutePath) async throws -> Report
+}
+
+public struct Rosalind: Rosalindable {
+    let fileSystem: FileSysteming
+    let commandRunner: CommandRunning
+
+    public init() {
+        self.init(fileSystem: FileSystem(), commandRunner: CommandRunner())
+    }
+
+    init(
+        fileSystem: FileSysteming,
+        commandRunner: CommandRunning
+    ) {
+        self.fileSystem = fileSystem
+        self.commandRunner = commandRunner
+    }
+
+    public func analyze(path: AbsolutePath) async throws -> Report {
+        if !(try await fileSystem.exists(path)) { throw RosalindError.notFound(path) }
+        return try await traverse(path: path, normalizingPath: path)
+    }
+
+    public func traverse(path: AbsolutePath, normalizingPath: AbsolutePath) async throws -> Report {
```

Comment:
```suggestion
    public func traverse(path: AbsolutePath, basePath: AbsolutePath) async throws -> Report {
```

`normalizingPath` feels a bit confusing in this context – to me, anyway